### PR TITLE
[CTX-601] chore: fix various AWS acceptance tests

### PR DESCRIPTION
**Pull request by craigfurman**
**chore: fix various AWS acceptance tests**

**chore: re-enable nightly acceptance tests**

This reverts commit 9907b1cd4245006d77670f66cd74fac4eaf2bf9c.

**chore: manual acceptance test job**

In order to trigger acceptance tests for certain branches pre-merge, and
without having to wait for a nightly build, you can click "trigger
pipeline" in CircleCI and enter `ACC_TESTS=1` as a parameter.

**chore: remove references to non-Snyk domains in acc test fixtures**

---

Attempt to fix failing AWS acceptance tests from https://app.circleci.com/pipelines/github/snyk/driftctl/5206/workflows/2425c0ab-f8a5-4594-8605-0e532e55ca88/jobs/13441/tests. Unless otherwise stated, I got a passing test against my sandbox account.

- TestAcc_Aws_ApiGatewayV2Authorizer, TestAcc_Aws_ApiGatewayAuthorizer: the nodejs 12 lambda runtime has been removed. Using a later one necessitated bumping the provider version. I bumped terraform too.
- TestAcc_Aws_LambdaEventSourceMapping: similar to above, bumped the lambda runtime, _but_ to a lower version to _avoid_ bumping the provider to a recent version. Driftctl is currently incompatible with new provider versions for this resource due to spurious attr-level drift on the destination_config block.
- TestAcc_Aws_CloudfrontDistribution, TestAcc_Aws_RDSCluster, TestAcc_Aws_RDSClusterInstance: deleted stray resources from previous failed acc test cleanups
- TestAcc_Aws_VPC, TestAcc_Aws_NetworkAcl_NonDeep, TestAcc_Aws_NetworkAcl: deleted stray resources. **I didn't actually run these tests, though**.
- TestAcc_Aws_S3Bucket_BucketInUsEast1: the most complicated one. The acl and policy attributes on a bucket are deprecated, and the actual non-deprecated policy failed to apply (HTTP 403 from AWS). I have no idea why. These attributes were added in https://github.com/snyk/driftctl/pull/471, and it's unclear from that PR why that was important to do. I removed all but one bucket from the test fixture, and removed all policy attachments to get this applying successfully at all. I would hope the relevant bucket policy attachment middlewares catch regressions here, but there's a chance this PR just encodes an already-occurred regression.

There are lots of formatting artifacts from my editor automatically terraform-fmting files, sorry!

### CODE REVIEW

File: pkg/resource/aws/testdata/acc/aws_lambda_event_source_mapping/main.tf [hunk 1]
```diff
@@ -88,11 +88,11 @@ resource "aws_iam_policy_attachment" "policy_attachment" {
...
-  filename         = "function.zip"
-  function_name    = "event-source-mapping-test-lambda-${local.timestamp}"
-  role             = aws_iam_role.iam_for_lambda.arn
-  handler          = "exports.test"
-  runtime       = "nodejs12.x"
+  filename      = "function.zip"
+  function_name = "event-source-mapping-test-lambda-${local.timestamp}"
+  role          = aws_iam_role.iam_for_lambda.arn
+  handler       = "exports.test"
+  runtime       = "nodejs14.x"
```
Comment (karniwl) [inline pkg/resource/aws/testdata/acc/aws_lambda_event_source_mapping/main.tf:95]: what's the reason this one is node 14 and line 89 at pkg/resource/aws/testdata/acc/aws_apigatewayv2_authorizer/terraform.tf is node 18? can't spot the difference and the original one are both the same version

**Review by karniwl — COMMENTED** [review]
left one small comment, but other than that, am I missing something or did we want to update the email to replace old cloudskiff ones?

**Comment by craigfurman** [comment]
@karniwl good call, I've pushed a commit to change those GCP fixtures just now. I haven't run those tests yet, but GCP acc tests are already broken - and are next on the list to fix! This should alleviate concerns around those IAM fixtures in the mean time.

**Review by karniwl — APPROVED** [review]
🧪 🧹

**Pull request merged**